### ISSUE TITLE
Support for multiple crowds

### ISSUE THREAD

**Issue by Ramshell** [TheSHEEEP/godotdetour#9]
Hi! First of all, thanks for the amazing work you've put into this project.

I've been digging a bit in the code, found that the navigation has only one crowd, and thought it'd be great to be able to use different crowds. The use case would be that for example in an RTS if you select a group of units then that's a crowd. At the same time, the enemy could have multiple groups of units going to different places. Do you think that'd be possible, or maybe I'm missing something?

**Comment by TheSHEEEP** [comment]
Unless I'm completely mistaken (and I might be, given that it's been a while since I wrote the plugin ;) ) - that should already be possible with only one crowd. I'd find it very strange if one crowd could only have all its agent walk towards the same target.

Though it isn't being done in the demo and I did not personally try it.

On the C++ side, when you call a function to set a move target for a detour crowd, you also have to supply the index of an agent to that function.

It shouldn't be too hard to change the demo in a way to test this - instead of setting the move target for every agent as is being done now, just set the move target for a single agent. That should lead to only that agent walking towards the move target, while the others continue doing what they did before.
